components/GistToProject.coffee

```diff
@@ -0,0 +1,60 @@
+noflo = require 'noflo'
+octo = require 'octo'
+path = require 'path'
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.inPorts.add 'in',
+    datatype: 'object'
+  c.outPorts.add 'graph',
+    datatype: 'object'
+  c.outPorts.add 'component',
+    datatype: 'object'
+  c.outPorts.add 'project',
+    datatype: 'object'
+  c.outPorts.add 'error',
+    datatype: 'object'
+
+  noflo.helpers.WirePattern c,
+    out: ['graph', 'component', 'project']
+    async: true
+  , (data, groups, out, callback) ->
+    api = octo.api()
+    api.token data.state.user['github-token'] if data.state.user?['github-token']
```

Comment:
Fair point, we can plug a `GetActionValues` in front of this so we only pick the pieces we need.